src/thermo.cpp

```diff
@@ -1134,8 +1134,8 @@ void Thermo::check_temp(const std::string &keyword)
   if (!temperature)
     error->all(FLERR, "Thermo keyword {} in variable requires thermo to use/init temperature",
                keyword);
-  if (update->first_update == 0)
-    error->all(FLERR,"Thermo keyword {} cannot be invoked before first run",keyword);
+  if (!temperature->is_initialized())
+    error->all(FLERR,"Thermo keyword {} cannot be invoked before initialized by run",keyword);
```

Comment:
"by a run" to be consistent with variable.cpp
ditto for other changes in this file